house-snark/src/relations/shielder/types.rs

```diff
@@ -122,12 +123,53 @@ mod casting {
 
     impl From<FrontendAccount> for BackendAccount {
         fn from(FrontendAccount(frontend_account): FrontendAccount) -> Self {
-            Self(CircuitField::new(BigInteger256::new([
-                u64::from_le_bytes(frontend_account[0..8].try_into().unwrap()),
-                u64::from_le_bytes(frontend_account[8..16].try_into().unwrap()),
-                u64::from_le_bytes(frontend_account[16..24].try_into().unwrap()),
-                u64::from_le_bytes(frontend_account[24..32].try_into().unwrap()),
-            ])))
+            Self(CircuitField::new(BigInteger256::new(bytes_to_4xu64(
+                &frontend_account,
+            ))))
+        }
+    }
+
+    //--------------------
+    // Byte representation
+    //--------------------
+
+    impl FrontendNote {
+        /// Create a note from the first 32 bytes of `bytes`.
+        pub fn from_bytes(bytes: &[u8]) -> FrontendNote {
+            FrontendNote(bytes_to_4xu64(bytes))
+        }
+
+        /// Serializes note to bytes.
+        pub fn to_bytes(self) -> Vec<u8> {
+            self.0
+                .iter()
+                .map(|elem| elem.to_le_bytes().to_vec())
+                .collect::<Vec<_>>()
+                .concat()
+        }
```

Comment:
These two _could_ be made into `From<AsRef<[u8]>> for FrontendNote` and `From<FrontendNote> for [u64; 4]`.

Reply:
that's right, but I didn't want to get also `into` counterparts from blanket implementation, so I used custom conversions